src/m365/pa/commands/app/app-consent-set.ts

```diff
@@ -0,0 +1,117 @@
+import { Cli } from '../../../../cli/Cli';
+import { Logger } from '../../../../cli/Logger';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import { powerPlatform } from '../../../../utils/powerPlatform';
+import { validation } from '../../../../utils/validation';
+import PowerAppsCommand from '../../../base/PowerAppsCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  environment: string,
+  name: string;
+  bypass: boolean;
+  confirm?: boolean;
+}
+
+class PaAppConsentSetCommand extends PowerAppsCommand {
+  public get name(): string {
+    return commands.APP_CONSENT_SET;
+  }
+
+  public get description(): string {
+    return 'Makes sure users can bypass the API Consent window for the selected canvas app';
+  }
+
+  constructor() {
+    super();
+
+    this.#initOptions();
+    this.#initValidators();
+    this.#initTypes();
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-e, --environment <environment>'
+      },
+      {
+        option: '-n, --name <name>'
+      },
+      {
+        option: '-b, --bypass <bypass>',
+        autocomplete: ['true', 'false']
+      },
+      {
+        option: '--confirm'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (!validation.isValidGuid(args.options.name)) {
+          return `${args.options.name} is not a valid GUID`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initTypes(): void {
+    this.types.boolean.push('bypass');
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    if (this.verbose) {
+      logger.logToStderr(`Setting the bypass consent for the Microsoft Power App ${args.options.name}... to ${args.options.bypass}`);
+    }
+
+    if (args.options.confirm) {
+      await this.consentPaApp(args);
+    }
+    else {
+      const result = await Cli.prompt<{ continue: boolean }>({
+        type: 'confirm',
+        name: 'continue',
+        default: false,
+        message: `Are you sure you bypass the consent for the Microsoft Power App ${args.options.name} to ${args.options.bypass}?`
+      });
+
+      if (result.continue) {
+        await this.consentPaApp(args);
+      }
+    }
+  }
+
+  private async consentPaApp(args: CommandArgs): Promise<void> {
+    const dynamicsApiUrl = await powerPlatform.getDynamicsInstanceApiUrl(args.options.environment);
+
+    const requestOptions: any = {
+      url: `${dynamicsApiUrl}/api/data/v9.0/canvasapps(${args.options.name})`,
```

Comment:
This endpoint throws some errors that some attributes are missing. It turns out that I can get a 200 OK if I execute it with the following: 

```json
data: {
  "bypassconsent": args.options.bypass,
  "name": args.options.name,
  "aadlastmodifiedbyid": "<my-user-object-guid>",
  "aadlastpublishedbyid": "<my-user-object-guid>"
}
```

but...it unfortunately does not work. If i execute an `m365 pa app get`, the bypassConsent property of the response remains the same as it was.

I tried using the PowerShell `Set-AdminPowerAppApisToBypassConsent` commandlet and that does work for me with the same parameters (environment and appName). I see the bypassConsent property changing when running the m365 pa app get. but still, I cannot return it to what it was using the CLI.

Did you get this to work before?

Also: if these extra properties are necessary, would that mean that we cannot run this in app only mode? If so we'd have to add a note to the docs.